### DISCUSSION TITLE
How to use get_pivots results - not memory locations

### DISCUSSION THREAD

**Discussion by anon2010** [DaveSkender/Stock.Indicators#942] [Help and support]
I'm passing in a dataframe with ohlcv data to get_pivots and I'm yet unable to figure out how to use/format the results. I just keep getting memory locations instead.

```
quotes_list = [
        Quote(d,o,h,l,c,v) 
        for d,o,h,l,c,v 
        in zip(pandas.to_datetime(df.time), df.open, df.high, df.low, df.close, df.volume)
    ]

pivots_list = indicators.get_pivots(quotes_list)

print()
print(pivots_list)
```

This prints:
`[<stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DDCAF7F0>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DDCAEC80>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DDCAF1F0>, <stock_indicators.indicators.pivots.PivotsResult object 
at 0x00000229DE1F00A0>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0100>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0160>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F01C0>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0220>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0280>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F02E0>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0340>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F03A0>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0400>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0460>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F04C0>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0520>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0580>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F05E0>, <stock_indicators.indicators.pivots.PivotsResult object at 0x00000229DE1F0640>,...`

I THINK I need to add something at the end of this line like
`in zip(pandas.to_datetime(df.time), df.open, df.high, df.low, df.close, df.volume).some-other-func()`

Any idea what I'm doing wrong here?

I thank you very much for the input.

**Comment by DaveSkender** [comment]
See the [basic example in our guide](https://python.stockindicators.dev/guide/#example-usage).  Essentially you're getting a series of `PivotsResult` class objects returned; you have to further specify which data element when using to depict values.

The response class for [Pivots](https://python.stockindicators.dev/indicators/Pivots/#content) is slightly more complex in that it also returns an `enum` that may translate different.  @LeeDongGeon1996 can explain better.

**Reply by anon2010** [reply]
Thank you very much for the input. I'm trying to get HH / LL into an existing pandas dataframe with ohlcv ("df").

```
    quotes_list = [
        Quote(d,o,h,l,c,v) 
        for d,o,h,l,c,v 
        in zip(pandas.to_datetime(df.time), df.open, df.high, df.low, df.close, df.volume)
    ]

    pivots_PivotsResult = indicators.get_pivots(quotes_list)

    # add HH and LL columns to df so that they can be filled when needed
    df['HH'] = ''
    df['LL'] = ''

    count = 0
    for r in pivots_PivotsResult:
        if r.high_trend == PivotTrend.HH:
            df['HH'].iloc[count] = '*'
        if r.low_trend == PivotTrend.LL:
            df['LL'].iloc[count] = '*'
        count = count + 1
```

For some reason, it looks like some of the results are accurate but then others are not. Obviously, this is due to my incorrect usage; not the indicator. Is this how you would go about putting this into an existing dataframe?

**Reply by LeeDongGeon1996** [reply]
I think you are using correctly. Can you give me more detail about it?
